### PR TITLE
Moved Application logic from DeveloperService to ApplicationService

### PR DESCRIPTION
**Pull request by RawToast**
* Moved FetchApplications and added a filter-free version
* Moved email list logic into Developers class
* Updated tests to reflect the changes

### CODE REVIEW

File: app/controllers/DevelopersController.scala [hunk 1]
```diff
@@ -85,14 +83,14 @@ trait DevelopersController extends FrontendController with GatekeeperAuthWrapper
...
-        apps <- developerService.fetchApplications(apiFilter)
+        apps <- applicationService.fetchApplications(apiFilter)
         apis <- apiDefinitionConnector.fetchAll
         filterOps = (developerService.filterUsersBy(apiFilter, apps) _
-                     andThen developerService.filterUsersBy(statusFilter) _)
-        devs <- developerService.fetchDevelopers
-        users = filterOps(devs)
-        emails = developerService.emailList(users)
-        page = PageableCollection(users, pageNumber, pageSize)
+                     andThen developerService.filterUsersBy(statusFilter))
```
Comment (rm-hull) [inline app/controllers/DevelopersController.scala]: Does this need to be partially applied ?

File: app/services/ApplicationService.scala [hunk 2]
```diff
@@ -17,8 +17,9 @@
 package services
 
 import connectors.ApplicationConnector
-import model.ResendVerificationSuccessful
+import model._
```
Comment (rm-hull) [inline app/services/ApplicationService.scala]: I thought the policy was to have explicit imports, not make them wide open?

File: app/services/DeveloperService.scala [hunk 3]
```diff
@@ -66,12 +49,11 @@ trait DeveloperService {
...
-  def emailList(users: Seq[User]) = {
-    val DELIMITER = "; " // Outlook requires email addresses separated by semi-colons
-    users.map(_.email).mkString(DELIMITER)
+  def fetchDevelopers(implicit hc: HeaderCarrier): Future[Seq[User]] = {
+    developerConnector.fetchAll.map(_.sorted)
   }
 
-  def fetchDevelopers(implicit hc: HeaderCarrier) = {
-    developerConnector.fetchAll.map(_.sorted)
+  def fetchAllDevelopers(implicit hc: HeaderCarrier): Future[Developers] = {
```
Comment (rm-hull) [inline app/services/DeveloperService.scala]: is this function used?

File: app/services/DeveloperService.scala [hunk 4]
```diff
@@ -66,12 +49,11 @@ trait DeveloperService {
...
+    developerConnector.fetchAll.map(users => Developers(users))
```
Comment (rm-hull) [inline app/services/DeveloperService.scala]: Do the users need to be sorted before passing int the wrapper?

File: app/model/Developers.scala [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+package model
+
+case class Developers(users: Seq[User]) {
+
+  private implicit val toDevelopers = (d: Seq[User]) => Developers(d)
+  private val DELIMITER = "; "
+
+  lazy val emailList = users.map(_.email).mkString(DELIMITER)
```
Comment (rm-hull) [inline app/model/Developers.scala]: I think the email list functionality has been removed entirely, so this line can probably go, and all the things that use it

File: app/model/Developers.scala [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+  def filterBy(filters: StatusFilter): Developers = {
```
Comment (rm-hull) [inline app/model/Developers.scala]: Is this method used?

Can't see that it is, so without the email list, not convinced of the value of the class

File: app/controllers/DevelopersController.scala [hunk 7]
```diff
@@ -85,14 +83,14 @@ trait DevelopersController extends FrontendController with GatekeeperAuthWrapper
...
+        allUsers <- developerService.fetchDevelopers
+        filteredUsers = filterOps(allUsers)
+        emails = Developers(filteredUsers).emailList
```
Comment (rm-hull) [inline app/controllers/DevelopersController.scala]: the email to button no longer exists, so this line should probably be removed along with any references

Comment (RawToast) [inline app/controllers/DevelopersController.scala]: This is still required for the "Copy to Clipboard" button.

File: app/model/Developers.scala (hunk 5 above)
Comment (RawToast) [inline app/model/Developers.scala]: As above, this is still required for the "Copy to Clipboard" button.

File: app/model/Developers.scala (hunk 6 above)
Comment (RawToast) [inline app/model/Developers.scala]: I was partially through pulling the filter logic out of the service, but stopped before things got messy. Will remove this for now.

**Pull request merged**